US National Archives Research Catalog.js

```diff
@@ -1,170 +1,160 @@
 }
 
+
 function doWeb(doc, url) {
-	var namespace = doc.documentElement.namespaceURI;
-	var nsResolver = namespace ? function(prefix) {
-		if (prefix == 'x') return namespace; else return null;
-	} : null;
+	var position = url.indexOf('/id/');
+	var id = url.substr(position+4);
+	var posturl = 'https://catalog.archives.gov/OpaAPI/iapi/v1/exports/noauth';
+	var postdata = 'export.format=json&export.type=full&export.what=metadata&naIds=' + id + '&rows=1';
 	
-	var articles = new Array();
-	
-	if (detectWeb(doc, url) == "multiple") {
-		var items = new Object();
-			
-		if (doc.evaluate('//div[@class="sT"]/p/strong[@class="sFC"]/a', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var titles = doc.evaluate('//div[@class="sT"]/p/strong[@class="sFC"]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);
+	ZU.doPost(posturl, postdata, function(result) {
+		var parsed = JSON.parse(result);
+		var exporturl = parsed.opaResponse.exportFile.url;
+		ZU.doGet(exporturl, function(data) {
+			var json = JSON.parse(data);
+			var item = new Zotero.Item("book");
 			
-		} else if (doc.evaluate('//td[3]/div[@class="sT"]/p/strong[@class="sFC"]/a', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext()) {
-			var titles = doc.evaluate('//td[3]/div[@class="sT"]/p/strong[@class="sFC"]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);
+			item.title = json[0].title;
+			var creators = json[0].creators;
+			for (var i=0; i<creators.length; i++) {
+				creators[i] = creators[i].replace('(Most Recent)', '');
+				if (creators[i].indexOf(", ")>-1) {
+					item.creators.push(ZU.cleanAuthor(creators[i], "author"));	
+				} else {
+					item.creators.push({'lastName': creators[i], 'creatorType': 'author'});
+				}
+			}
+			if (json[0].productionDates) {
+				item.date = json[0].productionDates[0];
+			} else {
+				item.date = json[0].date;
+			}
+			if (json[0].from) {
+				item.series = json[0].from[0];
+			}
+			item.abstractNote = json[0].scopeAndContentNote;
+			item.archive = json[0].archivedCopies.contacts1[0];
+			item.archiveLocation = json[0].localIdentifier;
+			item.extra = 'National Archives Identifier: ' + json[0].arcIdentifier;
 			
-		}	
-		
-		
+			item.attachments.push({
+				document: doc,
+				title: "Snapshot"
+			});
 
-		var next_title;
-		while (next_title = titles.iterateNext()) {
-			items[next_title.href] = next_title.textContent;
-		}
-		items = Zotero.selectItems(items);
-		for (var i in items) {
-			articles.push(i);
-		}
-	} else {
-		articles = [url];
+			item.complete();
+		})
+	});
+	
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://catalog.archives.gov/id/486076",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "The Struggle for Trade Union Democracy, December 1947",
+				"creators": [
+					{
+						"lastName": "Supreme Commander for the Allied Powers. Economic and Scientific Section. Director for Labor. Labor Division. 8/22/1949-4/28/1952 ",
```

Comment:
are we sure author fields of that length don't break sync? @dstillman -- what's the max length for creators in sync?

Can